rllib/utils/test_utils.py

```diff
@@ -372,6 +347,88 @@ def check_compute_single_action(trainer,
                         "({})!".format(action, what, action_space))
 
 
+def check_learning_achieved(tune_results, min_reward, evaluation=False):
+    """Throws an error if `min_reward` is not reached within tune_results.
+
+    Checks the last iteration found in tune_results for its
+    "episode_reward_mean" value and compares it to `min_reward`.
+
+    Args:
+        tune_results: The tune.run returned results object.
+        min_reward (float): The min reward that must be reached.
+
+    Raises:
+        ValueError: If `min_reward` not reached.
+    """
+    # Get maximum reward of all trials
+    # (check if at least one trial achieved some learning)
+    avg_rewards = [(trial.last_result["episode_reward_mean"]
+                    if not evaluation else
+                    trial.last_result["evaluation"]["episode_reward_mean"])
+                   for trial in tune_results.trials]
+    best_avg_reward = max(avg_rewards)
+    if best_avg_reward < min_reward:
+        raise ValueError("`stop-reward` of {} not reached!".format(min_reward))
+    print("ok")
+
+
+def check_train_results(train_results):
+    """Checks proper structure of a Trainer.train() returned dict.
+
+    Args:
+        train_results: The train results dict to check.
+
+    Raises:
+        AssertionError: If `train_results` doesn't have the proper structure or
+            data in it.
+    """
+
+    from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID
+    from ray.rllib.utils.metrics.learner_info import LEARNER_INFO, \
+        LEARNER_STATS_KEY
```

Comment:
is there a reason not to import these at the top of the file?
not important, just a style nit.

Reply:
Yeah, unfortunately, adding it to the top would lead to a circular import.

Reply:
hmm, that may be a sign that we should break up files / restructure things a little better.
we can do it later though, no need to block this PR.

Reply:
You are right. This typically happens when we have a structure like:

```
base_class_a.py -> imports something from utils
base_class_b.py -> imports something from utils

utils.py -> some simple utils, BUT also contains a useful function (e.g. for testing/debugging) that does need one of a or b
```

Reply:
oh ok, get it.
sounds like we just need to break utils.py into base_utils.py and utils.py. terrible at naming, you get the ideas.